src/client.ts

```diff
@@ -207,6 +209,36 @@ export default class Client implements ClientInterface {
     return this._state;
   }
 
+  private createDebugExecutable(env: NodeJS.ProcessEnv): Executable {
```

Comment:
Agreed. But I don't think we need any client options complexity, we should be able to just pass a flag to the ruby-lsp executable so that it does all the loading of the debug gem and setting up of the socket folders, etc. Something like `bundle exec ruby-lsp --debug` should be ideal.

Reply:
@st0012 we could also have it be another executable that is exported from the `ruby-lsp` package, like `ruby-lsp-debug`. I can't see how that might make it less secure than what we are doing here.